pkg/kubelet/cm/cpumanager/state/state_mem.go

```diff
@@ -77,33 +78,41 @@ func (s *stateMemory) SetCPUSet(podUID string, containerName string, cset cpuset
 	}
 
 	s.assignments[podUID][containerName] = cset
+	s.defaultCPUSet = defaultCPUSet
 	klog.InfoS("Updated desired CPUSet", "podUID", podUID, "containerName", containerName, "cpuSet", cset)
 }
 
-func (s *stateMemory) SetDefaultCPUSet(cset cpuset.CPUSet) {
+func (s *stateMemory) SetDefaultCPUSet(cset cpuset.CPUSet) error {
 	s.Lock()
 	defer s.Unlock()
 
+	if !s.defaultCPUSet.IsEmpty() {
+		return fmt.Errorf("Couldn't reset defaultCPUSet to: %v", cset)
+	}
+
 	s.defaultCPUSet = cset
 	klog.InfoS("Updated default CPUSet", "cpuSet", cset)
+	return nil
 }
 
-func (s *stateMemory) SetCPUAssignments(a ContainerCPUAssignments) {
+func (s *stateMemory) SetCPUAssignments(a ContainerCPUAssignments, defaultCPUSet cpuset.CPUSet) {
 	s.Lock()
 	defer s.Unlock()
 
 	s.assignments = a.Clone()
+	s.defaultCPUSet = defaultCPUSet
 	klog.InfoS("Updated CPUSet assignments", "assignments", a)
 }
 
-func (s *stateMemory) Delete(podUID string, containerName string) {
+func (s *stateMemory) Delete(podUID string, containerName string, defaultCPUSet cpuset.CPUSet) {
 	s.Lock()
 	defer s.Unlock()
 
 	delete(s.assignments[podUID], containerName)
 	if len(s.assignments[podUID]) == 0 {
 		delete(s.assignments, podUID)
 	}
+	s.defaultCPUSet = defaultCPUSet
```

Comment:
I'm a little curious. Why do we need to set this `defaultCPUSet`  value for the delete method?

Reply:
When deleting assignment, the defaultCPUSet should be updated. I think we want to write these data atomically, so the `Delete` function also should update defaultCPUSet.